api/v1alpha1/runner_types.go

```diff
@@ -28,10 +28,44 @@ type RunnerSpec struct {
 	Repository string `json:"repository"`
 
 	// +optional
-	Image string `json:"image"`
+	RunnerContainers []corev1.Container `json:"runnerContainers,omitempty"`
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+	// +optional
+	EnvFrom []corev1.EnvFromSource `json:"envFrom,omitempty"`
 
+	// +optional
+	Image string `json:"image"`
 	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// +optional
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+
+	// +optional
+	InitContainers []corev1.Container `json:"initContainers,omitempty"`
+	// +optional
+	Containers []corev1.Container `json:"containers,omitempty"`
```

Comment:
I prefer `SidecarContainers` for this one so that it becomes:

```
	if len(runner.Spec.SidecarContainers) != 0 {
		pod.Spec.Containers = append(pod.Spec.Containers, runner.Spec.SidecarContainers...)
	}
```

We're appending this to `pod.Spec.Containers` rather than replacing that, which makes me think this is better named after sidecars.

Reply:
I like this idea, thank you @mumoshu

Reply:
done